src/components/tools/image-component.tsx

```diff
@@ -10,7 +11,7 @@ interface IProps {
   onUrlChange: (url: string, filename?: string, context?: IDocumentContext) => void;
 }
 
-export const ImageComponent =
+const ImageComponent =
   React.forwardRef<HTMLDivElement, IProps>(({ content, style, onMouseDown, onUrlChange }, forwardedRef) => {
```

Comment:
It's not a huge deal, but I think you could have just wrapped this line in `observer()` and then you wouldn't have needed to define the separate `ObserverImageComponent` and change downstream clients. From an external perspective, `ImageComponent` is preferable.